deploy/docker/utils/bin/backup.test.js

```diff
@@ -0,0 +1,181 @@
+});
+
+it('Should not hould throw Error when the available size is >= MIN_REQUIRED_DISK_SPACE_IN_BYTES', () => {
+  expect(() => {backup.checkAvailableBackupSpace(Constants.MIN_REQUIRED_DISK_SPACE_IN_BYTES)}).not.toThrow('Not enough space avaliable at /appsmith-stacks. Please ensure availability of atleast 5GB to backup successfully.');
+});
+
+it('Generates t', async () => {
+  os.tmpdir =  jest.fn().mockReturnValue('temp/dir');
+  fsPromises.mkdtemp =  jest.fn().mockImplementation((a) => a);
+  backup.generateBackupRootPath().then((response)=>{console.log(response)})
+  const res = await backup.generateBackupRootPath()
+  expect(res).toBe('temp/dir/appsmithctl-backup-')
+});
+
+test('Test backup contents path generation', () => {
+  var root = '/rootDir'
+  var timestamp = '0000-00-0T00-00-00.00Z'
+  expect(backup.getBackupContentsPath(root, timestamp)).toBe('/rootDir/appsmith-backup-0000-00-0T00-00-00.00Z')
+});
+
+test('Test mongodump CMD generaton', async () => {
+  var dest = '/dest'
+  var appsmithMongoURI = 'mongodb://username:password@host/appsmith'
+  var cmd = 'mongodump --uri=mongodb://username:password@host/appsmith --archive=/dest/mongodb-data.gz --gzip'
+  utils.execCommand =  jest.fn().mockImplementation(async (a) => a.join(' '));
+  const res = await backup.executeMongoDumpCMD(dest, appsmithMongoURI)
+  expect(res).toBe(cmd)
+  console.log(res)
+})
+
+test('Test get gitRoot path when APPSMITH_GIT_ROOT is \'\' ', () => {
+  expect(backup.getGitRoot('')).toBe('/appsmith-stacks/git-storage')
+});
+
+test('Test get gitRoot path when APPSMITH_GIT_ROOT is null ', () => {
+  expect(backup.getGitRoot()).toBe('/appsmith-stacks/git-storage')
+});
+
+test('Test get gitRoot path when APPSMITH_GIT_ROOT is defined ', () => {
+  expect(backup.getGitRoot('/my/git/storage')).toBe('/my/git/storage')
+});
+
+test('Test ln command generation', async () => {
+  var gitRoot = '/appsmith-stacks/git-storage'
+  var dest = '/destdir'
+  var cmd = 'ln -s /appsmith-stacks/git-storage /destdir/git-storage'
+  utils.execCommand =  jest.fn().mockImplementation(async (a) => a.join(' '));
+  const res = await backup.executeCopyCMD(gitRoot, dest)
+  expect(res).toBe(cmd)
+  console.log(res)
+})
+
+it('Checks for the current Appsmith Version.', async () => {
+  
+  fsPromises.readFile =  jest.fn().mockImplementation(async (a) => 
+  `Object.defineProperty(exports, "__esModule", { value: true });
+  exports.VERSION = void 0;
+  exports.VERSION = "v0.0.0-SNAPSHOT";`);
+  const res = await backup.getCurrentVersion()
+  expect(res).toBe("v0.0.0-SNAPSHOT")
+  console.log(res)
+})
+
+test('If encriytpion env values are being removed', () => {
+  expect(backup.removeEncryptionEnvData(`APPSMITH_REDIS_URL=redis://127.0.0.1:6379\nAPPSMITH_ENCRYPTION_PASSWORD=hCzxbSpCCFSo4\nAPPSMITH_ENCRYPTION_SALT=hCzxbSpCCFSo4\nAPPSMITH_INSTANCE_NAME=Appsmith\n
```

Comment:
Let's make this explicit that this is a dummy password. 🙂

```suggestion
  expect(backup.removeEncryptionEnvData(`APPSMITH_REDIS_URL=redis://127.0.0.1:6379\nAPPSMITH_ENCRYPTION_PASSWORD=dummy-pass\nAPPSMITH_ENCRYPTION_SALT=dummy-salt\nAPPSMITH_INSTANCE_NAME=Appsmith\n
```